nemo_text_processing/text_normalization/en/verbalizers/money.py

```diff
@@ -63,55 +59,62 @@ def _get_minor_currencies(file):
-                + delete_space
-                + pynutil.delete("\"")
-                + ((pynini.closure(NEMO_NOT_QUOTE, 1) + minor_currencies_plural) | minor_currencies_singular)
+            fractional_non_one = (
+                pynutil.delete("fractional_part: \"")
+                + pynini.difference(pynini.closure(NEMO_NOT_QUOTE), pynini.union("oh one", "o one", "zero one", "one"))
                 + pynutil.delete("\"")
             )
-
-            # $2.00 {two zero zero dollars} -> two dollars
-            fractional_with_zeros = (
-                pynutil.delete("fractional_part:")
-                + delete_space
+            preserve_order = pynutil.delete("preserve_order: True")
+            unit_major_sing = pynini.string_file(get_abs_path("data/currency/currency.tsv"))
+            unit_major_plural = (
+                pynutil.delete("currency: \"")
+                + pynini.compose(unit_major_sing, SINGULAR_TO_PLURAL)
                 + pynutil.delete("\"")
-                + pynini.cross('zero', '')
-                + pynini.closure(pynini.cross(' zero', ''))
-                + delete_space
+            )
+            unit_major_sing = pynutil.delete("currency: \"") + unit_major_sing + pynutil.delete("\"")
+            unit_minor_sing = pynini.string_file(get_abs_path("data/currency/currency_minor_singular.tsv"))
+            unit_minor_sing = pynutil.delete("currency: \"") + unit_minor_sing + pynutil.delete("\"")
+            unit_minor_plural = pynini.string_file(get_abs_path("data/currency/currency_minor_plural.tsv"))
+            unit_minor_plural = pynutil.delete("currency: \"") + unit_minor_plural + pynutil.delete("\"")
+
+            integer_one = pynini.compose(decimal.integer, pynini.accep("one"))
+            integer_not_one = pynini.compose(decimal.integer, pynini.difference(NEMO_SIGMA, pynini.accep("one")))
+            graph_integer = integer_one + delete_space + insert_space + unit_major_sing + delete_space + preserve_order
+            graph_integer |= (
+                integer_not_one + delete_space + insert_space + unit_major_plural + delete_space + preserve_order
+            )
+            fractional_part_sing = (
+                delete_space
+                + pynutil.delete("fractional_part: \"" + pynini.union("o ", "oh ", "zero "))
+                + pynini.accep("one")
                 + pynutil.delete("\"")
                 + delete_space
+                + insert_space
+                + unit_minor_sing
             )
-
-            fractional = fractional_with_zeros | fractional_default
-
-            graph |= (
-                decimal.integer
+            graph_decimal_with_minor = (
+                graph_integer
                 + delete_space
                 + insert_space
-                + unit
+                + pynini.closure(pynutil.insert("and "), 0, 1)
+                + fractional_part_sing
+            )
+
+            fractional_part_plural = (
```

Comment:
could you add some comments with examples above each rule? otherwise after months we will not know what is what

Reply:
done